### ISSUE TITLE
Running babel-node from D:\ on Windows

### ISSUE THREAD

**Issue by krnlde** [babel/babel#1148]
I have my babel project on drive D:\ on Win8.1 and updated to babel 5.0.6 today. I now get this error message:

```
D:\project>babel-node --stage 0
> fs.js:546
  return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
                 ^
Error: ENOENT: no such file or directory, open 'D:\Users\[User]\.babel.json'
    at Error (native)
    at Object.fs.openSync (fs.js:546:18)
    at Object.fs.writeFileSync (fs.js:1149:15)
    at save (C:\Users\[User]\AppData\Roaming\npm\node_modules\babel\node_modules\ba
bel-core\lib\babel\api\register\cache.js:24:6)
    at process._tickDomainCallback (node.js:366:13)
    at Function.Module.runMain (module.js:453:11)
    at startup (node.js:123:18)
    at node.js:867:3
```

When I start babel-node from C:\ everything works fine since the `.babel.json` file is located correctly under `C:\Users\[User]\.babel.json`. I guess you guys forgot to check the absolute path starting _with_ the drive letter on windows.

**Comment by sebmck** [comment]
What's the result of running:

```sh
$ echo %USERPROFILE%
```

**Comment by krnlde** [comment]
`echo %USERPROFILE%` results in `C:\Users\[User]` correctly.

**Comment by sebmck** [comment]
This is weird. Also another reason to have [AppVeyor](http://www.appveyor.com/) in the CI, too bad the builds are super slow.

**Comment by sebmck** [comment]
This is the code that's used to resolve the `.babel.json` cache:

```javascript
function getUserHome() {
  return process.env.HOME || process.env.HOMEPATH || process.env.USERPROFILE;
}

const FILENAME = process.env.BABEL_CACHE_PATH || path.join(getUserHome() || os.tmpdir(), ".babel.json");
```

No real room for bugs. What's the result of running this in `node`:

```javascript
path.join(process.env.USERPROFILE, ".babel.json");
```

I currently don't have access to a Windows machine so I can't test very extensively.

**Comment by krnlde** [comment]
Running your example from node.js (io.js to be exact) it returns the correct path. I can't test it with babel-node for the reasons mentioned earlier :)

**Comment by krnlde** [comment]
Ah! Got something:

```
D:\quip.me>node
> process.env.HOME || process.env.HOMEPATH || process.env.USERPROFILE
'\\Users\\[User]'
>
```

After further investigation I found out `process.env.HOMEPATH === '\\Users\\[User]'`. Tada.

**Comment by sebmck** [comment]
Ah, great! I'll move `process.env.USERPROFILE` and push out a new release.

**Comment by krnlde** [comment]
Nice man! Thanks a lot

**Comment by krnlde** [comment]
@sebmck, when will you release 5.0.7?

**Comment by sebmck** [comment]
@krnlde Just trying to fix the Travis build so I don't push out a bad release.

**Comment by krnlde** [comment]
Ah OK! Sorry for the hustle :)

**Comment by AluisioASG** [comment]
Is there any reason for `HOMEPATH` being used at all?  According to [this ServerFault answer](https://serverfault.com/a/348323) `USERPROFILE` would be the semi-right place anyway (a better one would be `APPDATA`).
If you keep it you'll need to append it to either `HOMESHARE` or `HOMEDRIVE`, whichever is set (... `|| (process.env.HOMESHARE || process.env.HOMEDRIVE) + process.env.HOMEPATH`).

**Comment by sebmck** [comment]
@AluisioASG Pretty sure I just copy+pasted it from a StackOverflow answer.

**Comment by sebmck** [comment]
Fixed as of 5.0.7, thanks!

**Comment by krnlde** [comment]
Works like a charm now!
